### PR TITLE
Borrow sim

### PR DESCRIPTION
**Pull request by dpaiton**

### CODE REVIEW

File: elfpy/agents/agent.py [hunk 1]
```diff
@@ -254,13 +250,25 @@ def update_wallet(self, wallet_deltas: wallet.Wallet, market: Market) -> None:
...
             else:
                 raise ValueError(f"wallet_key={key} is not allowed.")
 
+    def _update_borrows(self, borrow_summary: wallet.Borrow) -> None:
+        if borrow_summary.start_time in self.wallet.borrows:  #  entry already exists for this mint_time, so add to it
+            self.wallet.borrows[borrow_summary.start_time].borrow_amount += borrow_summary.borrow_amount
+        else:
+            self.wallet.borrows.update({borrow_summary.start_time: borrow_summary})
+        if self.wallet.borrows[borrow_summary.start_time].borrow_amount == 0:
+            # Remove the empty borrow entry from the wallet.
```
Comment (sentilesdal) [inline elfpy/agents/agent.py:269]: i think this is an unnecessary comment, which i don't say often :P

Comment (sentilesdal) [reply elfpy/agents/agent.py:269]: if there is a problem that removing the empty one solves, that would be a better comment.

File: elfpy/agents/agent.py [hunk 2]
```diff
@@ -254,13 +250,25 @@ def update_wallet(self, wallet_deltas: wallet.Wallet, market: Market) -> None:
...
+                if value_or_dict:  # could be empty
+                    self._update_borrows(value_or_dict)
             elif key == "longs":
                 self._update_longs(value_or_dict.items())
             elif key == "shorts":
                 self._update_shorts(value_or_dict.items())
             else:
                 raise ValueError(f"wallet_key={key} is not allowed.")
 
+    def _update_borrows(self, borrow_summary: wallet.Borrow) -> None:
```
Comment (sentilesdal) [inline elfpy/agents/agent.py:263]: this looks good for now, i assume we'll be updating all fields in a borrow though, not just the borrow_amount?

File: elfpy/agents/agent.py [hunk 3]
```diff
@@ -423,28 +431,32 @@ def log_final_report(self, market: Market) -> None:
...
+                # " (%.2g in %s years), net worth = $%s"
+                # " from %s balance, %s longs, and %s shorts at p = %g\n"
             ),
             self.wallet.address,
             lost_or_made,
             profit_and_loss,
-            spend,
-            annual_percentage_rate,
-            holding_period_rate,
+            # spend,
```
Comment (sentilesdal) [inline elfpy/agents/agent.py:457]: ded code

**Review by sentilesdal — APPROVED** [review]
lgtm!

File: elfpy/agents/agent.py (hunk 2 above)
Comment (dpaiton) [reply elfpy/agents/agent.py:263]: yes eventually

**Pull request merged**